### ISSUE TITLE
Create testing sheet database or local test

### ISSUE THREAD

**Issue by andybaumgar** [nycmeshnet/nycmesh-support-bot#19]

**Comment by andybaumgar** [comment]
@Andrew-Dickinson I'm tempted to create a single drive API test and use a .csv with a mock drive client for the rest.  I think that will be more robust and centralized.  Creating a testing sheet database is probably easier though.  Any thoughts?

**Comment by Andrew-Dickinson** [comment]
I'd love to have an end-to-end test that uses a "beta" Google sheet (with fake data) for the lookup and the beta slack env for the messaging

**Comment by Andrew-Dickinson** [comment]
In general I think it would be good to get away from testing with real data

**Comment by andybaumgar** [comment]
Sounds good, I'll just make a test sheet for now then.  I had read some blog which recommended the drive client be modular so a local mock could be used instead for testing.  I'm curious to experiment though.

**Comment by andybaumgar** [comment]
Creating a testing dataset is turning into a while project.  For now I'm just going to write tests I don't commit.  Eventually the right way to do it is probably pulling real data and just anonymizing certain fields.  If we generate our own data from first principles I think we'll miss the most interesting cases.

**Comment by andybaumgar** [comment]
The number of strings used in the tests are also pretty minimal < 10, so we could always just add those to a testing .env file as well.

**Comment by andybaumgar** [comment]
I had a call with Andrew yesterday and he had a useful insight: developing the application requires "data science" which will naturally use the real data.  Once we commit the code into production the tests don't necessarily need that level of nuance, and we will know what values we can "blur out" in our testing dataset.  This will not make the tests less effective so long as the items we have blurred aren't important, or we have recreated them in anonymized form.
